tool/tctl/common/resource_command_test.go

```diff
@@ -902,54 +920,68 @@ func (test *dynamicResourceTest[T]) run(t *testing.T) {
 	// Fetch all resources.
 	buf, err = runResourceCommand(t, fileConfig, []string{"get", test.kind, "--format=json"})
 	require.NoError(t, err)
+	out = nil
 	mustDecodeJSON(t, buf, &out)
 	require.Len(t, out, 3)
-	require.Empty(t, cmp.Diff([]T{test.fooResource, test.fooBarResource, test.fooBarBazResource}, out,
+	require.Empty(t, cmp.Diff([]T{test.fooResource, test.fooBar1Resource, test.fooBar2Resource}, out,
 		cmpopts.IgnoreFields(types.Metadata{}, "ID", "Namespace"),
 	))
 
 	// Fetch specific resource.
 	buf, err = runResourceCommand(t, fileConfig,
 		[]string{"get", fmt.Sprintf("%v/%v", test.kind, test.fooResource.GetName()), "--format=json"})
 	require.NoError(t, err)
+	out = nil
 	mustDecodeJSON(t, buf, &out)
 	require.Len(t, out, 1)
 	require.Empty(t, cmp.Diff([]T{test.fooResource}, out,
 		cmpopts.IgnoreFields(types.Metadata{}, "ID", "Namespace"),
 	))
 
 	// Remove a resource.
-	_, err = runResourceCommand(t, fileConfig, []string{"rm", fmt.Sprintf("%v/%v", test.kind, test.fooBarResource.GetName())})
+	_, err = runResourceCommand(t, fileConfig, []string{"rm", fmt.Sprintf("%v/%v", test.kind, test.fooResource.GetName())})
 	require.NoError(t, err)
 
 	// Fetch all resources again.
 	buf, err = runResourceCommand(t, fileConfig, []string{"get", test.kind, "--format=json"})
 	require.NoError(t, err)
+	out = nil
 	mustDecodeJSON(t, buf, &out)
 	require.Len(t, out, 2)
-	require.Empty(t, cmp.Diff([]T{test.fooResource, test.fooBarBazResource}, out,
+	require.Empty(t, cmp.Diff([]T{test.fooBar1Resource, test.fooBar2Resource}, out,
 		cmpopts.IgnoreFields(types.Metadata{}, "ID", "Namespace"),
 	))
 
-	if !test.runPrefixNameChecks {
+	if !test.runDiscoveredNameChecks {
 		return
 	}
 
-	// Test prefix name behavior.
-	// Removing multiple resources ("foo" and "foo-bar-baz")by prefix name is an error.
-	_, err = runResourceCommand(t, fileConfig, []string{"rm", fmt.Sprintf("%v/%v", test.kind, "f")})
+	// Test discovered name behavior.
+	// Fetching multiple resources ("foo-bar-1" and "foo-bar-2") by discovered name is ok.
+	buf, err = runResourceCommand(t, fileConfig, []string{"get", fmt.Sprintf("%v/%v", test.kind, "foo-bar"), "--format=json"})
+	require.NoError(t, err)
+	out = nil
+	mustDecodeJSON(t, buf, &out)
```

Comment:
nit: maybe make a new helper to do the reset then decode

Reply:
Refactored the yaml/json decoders to do exactly that: 3adaf7d330c7b0188d8275bbed05b437b5710f6c

I realized while reworking the tests that when the var is already decoded into it should be reset before reuse, so this prevents that mistake from happening again